core/src/main/java/uk/ac/bbsrc/tgac/miso/core/security/util/LimsSecurityUtils.java

```diff
@@ -88,8 +89,11 @@ public static void updateFromLdapUser(User target, InetOrgPerson ldapUserDetails
     target.setInternal(roles.contains(MisoAuthority.ROLE_INTERNAL.name()));
 
     target.setPassword(ldapUserDetails.getPassword());
-    target.setFullName(ldapUserDetails.getDisplayName());
-    target.setEmail(ldapUserDetails.getMail());
+    if(ldapUserDetails instanceof InetOrgPerson) {
+      InetOrgPerson person = (InetOrgPerson) ldapUserDetails;
+      target.setFullName(person.getDisplayName());
+      target.setEmail(person.getMail());
+    }
```

Comment:
Will fail to save because fullname isn't nullable. Email is nullable in the DB, but I'm not sure that's intentional as it's required on the Edit User page and there previously hasn't been a way for it to be null. You'll have to check the call hierarchy to see if null email would cause any problems. Maybe you could just use the username as the fullname too, but I'm not sure there's any way to get an email

Reply:
Using the username as full name works. Email is harder since UserDetails doesn't assume an email address is associated with a user. Searching for 'email' across all of MISO it looks as though it's tracked but no actual features use it? Should we make emails optional?